src/Compilers/CSharp/Portable/Symbols/Source/SourceOrdinaryMethodSymbol.cs

```diff
@@ -758,98 +483,10 @@ internal override bool IsExpressionBodied
-                }
-            }
-            else if (isInterface)
-            {
-                Debug.Assert(isExplicitInterfaceImplementation);
-                allowedModifiers |= DeclarationModifiers.Abstract;
-            }
-
-            allowedModifiers |= DeclarationModifiers.Extern | DeclarationModifiers.Async;
-
-            if (ContainingType.IsStructType())
-            {
-                allowedModifiers |= DeclarationModifiers.ReadOnly;
-            }
-
-            // In order to detect whether explicit accessibility mods were provided, we pass the default value
-            // for 'defaultAccess' and manually add in the 'defaultAccess' flags after the call.
-            bool hasExplicitAccessMod;
-            var mods = ModifierUtils.MakeAndCheckNontypeMemberModifiers(modifiers, defaultAccess: DeclarationModifiers.None, allowedModifiers, location, diagnostics, out bool modifierErrors);
-            if ((mods & DeclarationModifiers.AccessibilityMask) == 0)
-            {
-                hasExplicitAccessMod = false;
-                mods |= defaultAccess;
-            }
-            else
-            {
-                hasExplicitAccessMod = true;
-            }
-
-            this.CheckUnsafeModifier(mods, diagnostics);
-
-            ModifierUtils.ReportDefaultInterfaceImplementationModifiers(hasBody, mods,
-                                                                        defaultInterfaceImplementationModifiers,
-                                                                        location, diagnostics);
-
-            mods = AddImpliedModifiers(mods, isInterface, methodKind, hasBody);
-            return (mods, hasExplicitAccessMod, modifierErrors);
-        }
-
-        private static DeclarationModifiers AddImpliedModifiers(DeclarationModifiers mods, bool containingTypeIsInterface, MethodKind methodKind, bool hasBody)
+        protected override DeclarationModifiers MakeDeclarationModifiers(DeclarationModifiers allowedModifiers, DiagnosticBag diagnostics)
         {
-            // Let's overwrite modifiers for interface and explicit interface implementation methods with what they are supposed to be. 
-            // Proper errors must have been reported by now.
-            if (containingTypeIsInterface)
-            {
-                mods = ModifierUtils.AdjustModifiersForAnInterfaceMember(mods, hasBody,
-                                                                         methodKind == MethodKind.ExplicitInterfaceImplementation);
-            }
-            else if (methodKind == MethodKind.ExplicitInterfaceImplementation)
-            {
-                mods = (mods & ~DeclarationModifiers.AccessibilityMask) | DeclarationModifiers.Private;
-            }
-            return mods;
+            var syntax = GetSyntax();
+            return ModifierUtils.MakeAndCheckNontypeMemberModifiers(syntax.Modifiers, defaultAccess: DeclarationModifiers.None, allowedModifiers, Locations[0], diagnostics, out _);
```

Comment:
nit: consider naming discarded argument

Reply:
> nit: consider naming discarded argument

I think I intentionally made it discard.

---
In reply to: [445753835](https://github.com/dotnet/roslyn/pull/45424#discussion_r445753835) [](ancestors = 445753835)